packages/diffhtml-middleware-worker/lib/create-web-socket.js

```diff
@@ -0,0 +1,30 @@
+import diff from './util/binding';
+
+const { innerHTML, html } = diff;
+const { parse } = JSON;
+
+export const createWebSocket = (socketUrl, { mount, socketOptions }) => {
+  const socket = new WebSocket(socketUrl, socketOptions);
+
+  socket.addEventListener('message', async e => {
+    const { type, ...rest } = parse(e.data);
+
+    // TODO Deprecate the 'clear' event. This is currently used in the Node
+    // worker when an error is encountered. This cleans up the markup to avoid
+    // issues during rendering.
+    if (type === 'clear') {
```

Comment:
Need  to come up with a spec to define which types are handled. Should additionally namespace the type to something like `diff:clear`, `diff:patches`, etc.